src/Widgets/TaskRow.vala

```diff
@@ -67,25 +66,97 @@ public class Tasks.TaskRow : Gtk.ListBoxRow {
         grid.attach (summary_label, 1, 0);
         grid.attach (description_grid, 1, 1);
 
-        if ( !ical_task.get_due ().is_null_time () ) {
+        due_label = new Gtk.Label (null);
+        unowned Gtk.StyleContext due_label_context = due_label.get_style_context ();
+        due_label_context.add_provider (taskrow_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+        due_label_context.add_class ("due-date");
+        description_grid.add (due_label);
+
+        description_label = new Gtk.Label (null);
+        description_label.xalign = 0;
+        description_label.lines = 1;
+        description_label.ellipsize = Pango.EllipsizeMode.END;
+        description_label.get_style_context ().add_class (Gtk.STYLE_CLASS_DIM_LABEL);
+        description_grid.add (description_label);
+
+        var eventbox = new Gtk.EventBox ();
+        eventbox.expand = true;
+        eventbox.above_child = false;
+        eventbox.add (grid);
+
+        add (eventbox);
+
+        eventbox.event.connect ((event) => {
+            if (event.type == Gdk.EventType.@2BUTTON_PRESS) {
+                var task_popover = new Tasks.TaskSettingsPopover (task);
+                task_popover.constrain_to = Gtk.PopoverConstraint.NONE;
+                task_popover.position = Gtk.PositionType.LEFT;
+                task_popover.set_relative_to (check);
+
+                task_popover.closed.connect (() => {
+                    task_changed (task);
+                });
+
+                task_popover.popup ();
+
+            } else if (!is_selected () && event.type == Gdk.EventType.BUTTON_PRESS) {
+                activate ();
+            }
+        });
+
+        check.toggled.connect (() => {
+            if (task == null) {
+                return;
+            }
+            task.get_icalcomponent ().set_status (check.active ? ICal.PropertyStatus.COMPLETED : ICal.PropertyStatus.NONE);
+            task_changed (task);
+        });
+
+        key_release_event.connect ((event) => {
+            // SPACE == 32
+            if (event.keyval == 32) {
+                check.active = !check.active;
+            }
+        });
+    }
+
+    public void update_request (ECal.Component task) {
+        this.task = task;
+
+        unowned ICal.Component ical_task = task.get_icalcomponent ();
+        completed = ical_task.get_status () == ICal.PropertyStatus.COMPLETED;
+        check.active = completed;
+
+        summary_label.label = ical_task.get_summary ();
+
+        if (completed) {
+            summary_label.get_style_context ().add_class (Gtk.STYLE_CLASS_DIM_LABEL);
+            description_grid.get_style_context ().add_class (Gtk.STYLE_CLASS_DIM_LABEL);
+        } else {
+            summary_label.get_style_context ().remove_class (Gtk.STYLE_CLASS_DIM_LABEL);
+            description_grid.get_style_context ().remove_class (Gtk.STYLE_CLASS_DIM_LABEL);
+        }
+
+        if ( ical_task.get_due ().is_null_time () ) {
+            due_label.visible = false;
```

Comment:
visible property has the problem that `show_all ()` overrides it. In general it's better to just use [Gtk.Revealer](https://valadoc.org/gtk+-3.0/Gtk.Revealer.html). Especially because you get nice transitions and we want that anyways

Reply:
Understood. Will change the Code accordingly. Stumbled across Gtk.Revealer before and was wondering what it might be doing anyways ;)

Reply:
Implemented the suggested change. However, it seems I still need to set the `due_label_revealer`'s visibility to `false` after it has hidden it's child in order to free the space and allow the description to span over the whole row.

I hope I did this the way suggested. If there's a better way to do this I happily apply it.